### ISSUE TITLE
Magento Error Array To String Conversion

### ISSUE THREAD

**Issue by PatrickKluwig** [develodesign/vsf-payment-stripe#17]
Hey, when I want to pay with stripe Account, I got the following error in my Magento 2 log:
`main.CRITICAL: Report ID: webapi-5cd423334fe8e; Message: Notice: Array to string conversion in /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Reflection/TypeProcessor.php on line 473 {"exception":"[object] (Exception(code: 0): Report ID: webapi-5cd423334fe8e; Message: Notice: Array to string conversion in /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Reflection/TypeProcessor.php on line 473 at /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Webapi/ErrorProcessor.php:206, Exception(code: 0): Notice: Array to string conversion in /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Reflection/TypeProcessor.php on line 473 at /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/App/ErrorHandler.php:61)"} []
`

**Comment by 99percent6** [comment]
+1
have same problem
it's happend when we send data for create order to magento

**Comment by rain2o** [comment]
I did some debugging after receiving this same error and this is what I have found. 

The error we see about Array to String conversion is because the expected schema of the POST body when calling Magento's API is the following:
```
{
  "paymentMethod": {
    "po_number": "string",
    "method": "string",
    "additional_data": [
      "string"
    ],
    "extension_attributes": {
      "agreement_ids": [
        "string"
      ]
    }
  }
}
```

Notice in `additional_data` it is expecting an array of strings only, but the data being sent contains nested objects inside `additional_data`. Specifically the problem is the `card` object. 

So, I was curious how the Magento 2 Stripe module handles this on Magento front-end, and as a result how it expects the data in the payment method module. What I found is they actually extract three fields from the `card` object and send the data as a concatenated string using colon ":" as a separator. Below is an example of what the M2 Stripe module sends in `additional_data` by default. This was from testing the M2 Stripe module in Magento's front-end on the Luma theme.
```
"additional_data": {
    "cc_stripejs_token": "pm_2EzJJAJ7x7C9MXXFDsim7WW1:visa:4242",
    "cc_save": false
}
```
The value for `cc_stripejs_token` was created using the following code (from `/vendor/stripe/module-payments/view/frontend/web/js/view/payment/method-renderer/stripe_payments.js#116`):
```
token.id + ':' + token.card.brand + ':' + token.card.last4
```

In contrast, currently the data being sent from this VSF module is something like this.
```
"additional_data": {
    "id": "tok_2Ez5xTJ7x7C9MXXF8C4TnRvH",
    "object": "token",
    "card": {
        "id": "card_2Ez5xTJ7x7C9MXXFvoiuyuod",
        "object": "card",
        "address_city": null,
        "address_country": null,
        "address_line1": null,
        "address_line1_check": null,
        "address_line2": null,
        "address_state": null,
        "address_zip": null,
        "address_zip_check": null,
        "brand": "Visa",
        "country": "FR",
        "cvc_check": "unchecked",
        "dynamic_last4": null,
        "exp_month": 12,
        "exp_year": 2022,
        "funding": "credit",
        "last4": "0003",
        "metadata": {},
        "name": null,
        "tokenization_method": null
    },
    "client_ip": "...",
    "created": 1563818999,
    "livemode": false,
    "type": "card",
    "used": false
}
```

This means in order to be compatible with Stripe's official M2 module, this module will need to send the data in the same format. However, what I don't know is if this specific format is used in all platforms where Stripe provides an official payment method plugin/module/extension/whatever.

The more convenient place to make this change is probably https://github.com/develodesign/vsf-payment-stripe/blob/master/components/PaymentStripe.vue#L116. But if that is done then there will need to be logic in place to account for which platform is being used, and handle according to its specific requirements.

Otherwise this data could be parsed directly in [vue-storefront-api/src/platform/magento2/o2m.js](https://github.com/DivanteLtd/vue-storefront-api/blob/master/src/platform/magento2/o2m.js), but again it would need logic in there to only do this formatting if it's the Stripe payment method. This is probably not the best as it would require modifying VSF API core to account for possible Stripe.
